apis/nucleus/src/components/listbox/components/ListBoxHeader/icon-utils.js

```diff
@@ -0,0 +1,36 @@
+import DrillDownIcon from '@nebula.js/ui/icons/drill-down';
+import CyclicIcon from '@nebula.js/ui/icons/reload';
+
+const dimensionTypes = {
+  single: 'N',
+  drillDown: 'H',
+  cyclic: 'C',
+};
+
+const createDimensionIconData = (dimInfo, app) => {
+  switch (dimInfo.qGrouping) {
+    case dimensionTypes.drillDown:
+      return {
+        icon: DrillDownIcon,
+        tooltip: 'Listbox.DrillDown',
+        onClick: undefined,
+      };
+    case dimensionTypes.cyclic:
+      return {
+        icon: CyclicIcon,
+        tooltip: 'Listbox.Cyclic',
+        onClick: () => {
+          app
+            .getDimension(dimInfo.qLibraryId)
+            .then((dimensionModel) => {
+              dimensionModel.stepCycle(1);
+            })
+            .catch(() => null);
+        },
+      };
+    default:
+      return undefined;
+  }
+};
+
+export default { createDimensionIconData };
```

Comment:
Is it intentional to export it with default?

Typically, I would otherwise do:
```
export const createDimensionIconData(…) => {
```

and then either:
```
import { createDimensionIconData } from './icon-utils';
```
or
```
import * as iconUtils from './icon-utils';
```

Reply:
It is intentional since it is only one function I can only du `export default function...` but I don't like that because "cosummer" could write anything like `import anyName from "./icon-utils;` and then use `anyName()`. So it might not show up when you search for the function by its original name. But if you think I should change it then I will change since I usually not working with this repo and don't know the common rules in it.